--- glib-1.2.10/glib.h~	2019-10-23 08:41:07.000000000 +0200
+++ glib-1.2.10/glib.h	2019-10-23 08:41:33.443203482 +0200
@@ -528,7 +528,7 @@
 	 if (__builtin_constant_p (val))		\
 	   __v = GUINT16_SWAP_LE_BE_CONSTANT (val);	\
 	 else						\
-	   __asm__ __const__ ("rorw $8, %w0"		\
+	   __asm__ ("rorw $8, %w0"		\
 			      : "=r" (__v)		\
 			      : "0" ((guint16) (val)));	\
 	__v; }))
@@ -541,7 +541,7 @@
 	    if (__builtin_constant_p (val))			\
 	      __v = GUINT32_SWAP_LE_BE_CONSTANT (val);		\
 	  else							\
-	    __asm__ __const__ ("rorw $8, %w0\n\t"		\
+	    __asm__ ("rorw $8, %w0\n\t"		\
 			       "rorl $16, %0\n\t"		\
 			       "rorw $8, %w0"			\
 			       : "=r" (__v)			\
@@ -554,7 +554,7 @@
 	    if (__builtin_constant_p (val))			\
 	      __v = GUINT32_SWAP_LE_BE_CONSTANT (val);		\
 	  else							\
-	    __asm__ __const__ ("bswap %0"			\
+	    __asm__ ("bswap %0"			\
 			       : "=r" (__v)			\
 			       : "0" ((guint32) (val)));	\
 	__v; }))
